### PR TITLE
HomeWork 1

### PR DESCRIPTION
**Pull request by iruzevic**
### Added
- Local storage helpers
- Initial HTML, CSS files
- JS functionality
- star rating

# Preview

## With Input
https://github.com/infinum-academy/js-react-academy-2024-iruzevic/assets/23283324/2c20a4f1-94de-4449-a299-7f988198b7aa

## With Rating stars
https://github.com/infinum-academy/js-react-academy-2024-iruzevic/assets/23283324/8cd8f8b1-c9b2-4d9b-a4f4-718547005644

## Note
If you open the HTML file directly in the browser, the Js will not work due to the `module` being used for easier imports. You can use the `https://www.npmjs.com/package/serve` package for a simple server, and all is good.

### CODE REVIEW

File: project/index.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+ * Calculate the average rating and update the overview element.
+ *
+ * @returns {void}
+ */
+function calcOverview() {
+	let totalRating = 0;
+
+	const reviews = getLocalStorage(storageName);
+
+	reviews.forEach((review) => totalRating += review.rating);
```
Comment (veksi) [inline project/index.js:108]: Zašto ne zakomplicirati život 😄
```suggestion
	const totalRating = reviews.reduce((acc, review) => {
		return acc + parseInt(review.rating);
	}, 0) / reviews.length;
```

File: project/index.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	const ratingItemButton = clone.querySelector('.js-rating-template-delete');
+	const ratingItemRating = clone.querySelector('.js-rating-template-rating');
+
+	ratingItemContent.textContent = content;
+	ratingItemValue.textContent = rating;
+	ratingItemRating.setAttribute('data-rating', rating);
+
+	ratingItemButton.onclick = () => {
+		reRenderReviews(
+			getLocalStorage(storageName).filter((t) => JSON.stringify(t) !== JSON.stringify(review))
```
Comment (veksi) [inline project/index.js]: Ovo neće raditi u slučaju da imaš iste reviewove (isti rating i content), tj. obrisat će sve iste ratinge a ne samo onaj koji si kliknio

File: project/styles.css [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	grid-column: span 12;
+	color: var(--color-white);
+	font-size: 30px;
+	cursor: pointer;
+	display: grid;
+	grid-template-columns: repeat(12, 1fr);
+	gap: 5px;
+}
+
+.rating-stars[data-rating='1'] > :nth-child(-n+1),
```
Comment (veksi) [inline project/styles.css:138]: Trebalo mi je malo da skuzim sta je pjesnik htio reći ali 👍 .

**Review by veksi — CHANGES_REQUESTED** [review]
Almost perfect, 99% :D 
Nešto sam i naučio iz ovog, template element 👍  (vidi se da radis WP)
Postoji bug u brisanju elemenata samo.

**Review by veksi — APPROVED** [review]
Pls samo rješiš onaj bug sa brisanjem i GG!

File: project/index.js (hunk 2 above)
Comment (iruzevic) [reply project/index.js]: done, tnx

**Comment by iruzevic** [comment]
All done

**Review by veksi — APPROVED** [review]
❤️

**Pull request merged**